source/acquaman/BioXAS/BioXASMainXASScanActionController.cpp

```diff
@@ -99,13 +99,35 @@ AMAction3* BioXASMainXASScanActionController::createInitializationActions()
     stage4->addSubAction(scaler->createStartAction3(true));
     stage4->addSubAction(scaler->createWaitForDwellFinishedAction(regionTime + 5.0));
 
+    // setup beamline for dark current correction.
+
+    AMListAction3 *darkCurrentSetup = new AMListAction3(new AMListActionInfo3("BioXAS Main Dark Current Setup", "BioXAS Main Dark Current Setup"), AMListAction3::Parallel);
+
+    for (int i = 0, size = configuration_->detectorConfigurations().count(); i < size; i++) {
+	    AMDetector *detector = AMBeamline::bl()->exposedDetectorByInfo(configuration_->detectorConfigurations().at(i));
+	    bool sharedSourceFound = false;
+
+	    if (detector) {
+		    int detectorIndex = scan_->indexOfDataSource(detector->name());
+
+		    // if we have a valid detector that can perform dark current correction, make that detector perform a dark current measurement.
+		    if (detectorIndex != -1 && detector->rank() == 0 && detector->canDoDarkCurrentCorrection()) {
+			    bool sourceShared = detector->sharesDetectorTriggerSource();
+
+			    if ((sourceShared && !sharedSourceFound) || !sourceShared)
```

Comment:
Also, using it here after it is never changed unnecessarily complicates this if statement.